### ISSUE TITLE
Parser fails to detect episodes correctly

### ISSUE THREAD

**Issue by TTU-t** [Sonarr/Sonarr#8728]
### Is there an existing issue for this?

- [x] I have searched the existing open and closed issues

### Current Behavior

For this torrent, episodes 1 - 14 are detected incorrectly, but episodes after 15 work fine:

<img width="1360" height="814" alt="Image" src="https://github.com/user-attachments/assets/459a0172-c6bc-4d5e-847e-ad403a8663cd" />

I'm not really sure why as the episodes seem to be labelled correctly.

### Expected Behavior

Correctly assigned episodes as follows:

<img width="1335" height="539" alt="Image" src="https://github.com/user-attachments/assets/37bcd9bc-90ab-43ac-8691-d84e692a0954" />

### Steps To Reproduce

I'm not sure what else youd like me to provide

### Environment

```markdown
- OS: Ubuntu
- Sonarr: 4.0.17.2952
- Docker Install: Yes
- Using Reverse Proxy: No
```

### What branch are you running?

Main

### Trace Logs?

Trace extract for one episode:

> 2026-06-20 12:20:29.6|Debug|Parser|Parsing string 'Courage the Cowardly Dog - S03E09-E10 - Scuba Scuba Doo & Conway the Contaminationist (1080p x265 EDGE2020).mkv'
2026-06-20 12:20:29.6|Debug|Parser|Episode Parsed. Courage the Cowardly Dog - S03E09-10 
2026-06-20 12:20:29.6|Debug|Parser|Languages parsed: Unknown
2026-06-20 12:20:29.6|Debug|QualityParser|Trying to parse quality for 'Courage the Cowardly Dog - S03E09-E10 - Scuba Scuba Doo & Conway the Contaminationist (1080p x265 EDGE2020).mkv'
2026-06-20 12:20:29.6|Debug|Parser|Quality parsed: HDTV-1080p v1
2026-06-20 12:20:29.6|Debug|Parser|Release Group parsed: EDGE2020
2026-06-20 12:20:29.6|Debug|VideoFileInfoReader|Getting media info from /Storagge/Downloads/sonarr-imp/Courage the Cowardly Dog (1999) Season 1-4 S01-04 (1080p HMAX WEBDL x265 10bit AAC 2.0 EDGE2020)/Season 3/Courage the Cowardly Dog - S03E09-E10 - Scuba Scuba Doo & Conway the Contaminationist (1080p x265 EDGE2020).mkv
2026-06-20 12:20:29.6|Debug|Parser|Parsing string 'Courage the Cowardly Dog - S03E09-E10 - Scuba Scuba Doo & Conway the Contaminationist (1080p x265 EDGE2020)'
2026-06-20 12:20:29.6|Debug|ParsingService|Using Scene to TVDB Mapping for: Courage the Cowardly Dog - Scene: 3x09 - TVDB: 3x17, 3x18
2026-06-20 12:20:29.6|Debug|ParsingService|Using Scene to TVDB Mapping for: Courage the Cowardly Dog - Scene: 3x10 - TVDB: 3x19, 3x20
2026-06-20 12:20:29.6|Debug|AggregateLanguage|Selected languages: English
2026-06-20 12:20:29.6|Debug|AggregateQuality|Using quality: WEBDL-1080p v1
2026-06-20 12:20:29.6|Debug|AbsoluteEpisodeNumberSpecification|Series type is not Anime, skipping check
2026-06-20 12:20:29.6|Debug|AlreadyImportedSpecification|No download client information is available, skipping
2026-06-20 12:20:29.6|Debug|EpisodeTitleSpecification|Episode aired more than 48 hours ago
2026-06-20 12:20:29.6|Debug|EpisodeTitleSpecification|Episode aired more than 48 hours ago
2026-06-20 12:20:29.6|Debug|EpisodeTitleSpecification|Episode aired more than 48 hours ago
2026-06-20 12:20:29.6|Debug|EpisodeTitleSpecification|Episode aired more than 48 hours ago
2026-06-20 12:20:29.6|Debug|Parser|Parsing string 'Courage the Cowardly Dog - S03E09-E10 - Scuba Scuba Doo & Conway the Contaminationist (1080p x265 EDGE2020)'
2026-06-20 12:20:29.6|Debug|ParsingService|Using Scene to TVDB Mapping for: Courage the Cowardly Dog - Scene: 3x09 - TVDB: 3x17, 3x18
2026-06-20 12:20:29.6|Debug|ParsingService|Using Scene to TVDB Mapping for: Courage the Cowardly Dog - Scene: 3x10 - TVDB: 3x19, 3x20
2026-06-20 12:20:29.6|Debug|MatchesFolderSpecification|Unexpected episode(s) in file: 3x17, 3x18, 3x19, 3x20
